homeassistant/components/climate/fritzbox.py

```diff
@@ -130,9 +130,9 @@ def operation_list(self):
 
     def set_operation_mode(self, operation_mode):
         """Set new operation mode."""
-        if operation_mode == STATE_HEAT:
+        if operation_mode in (STATE_HEAT, STATE_AUTO):
```

Comment:
These new modes aren't returned by current operation either, making it very weird. We ask it to set a mode, and it changes to another mode.